daemon/volumes.go

```diff
@@ -164,11 +177,25 @@ func (container *Container) registerVolumes() {
 		if rw, exists := container.VolumesRW[path]; exists {
 			writable = rw
 		}
+		runtime.LockOSThread()
+		defer resetLabeling()
+
+		if err := label.SetFileCreateLabel(container.MountLabel); err != nil {
+			logrus.Errorf("Unable to setup default labeling for volume creation %s: %v", path, err)
+			continue
+
+		}
+
+		// Create the actual volume
 		v, err := container.daemon.volumes.FindOrCreateVolume(path, writable)
 		if err != nil {
 			logrus.Debugf("error registering volume %s: %v", path, err)
 			continue
 		}
+		if err := resetLabeling(); err != nil {
+			logrus.Debugf("Unable to reset labeling %s: %v", path, err)
```

Comment:
I'm personally think that `resetLabeling` should panic on error :) But at least `Errorf` will be okay for me. Also in `defer` too :)